libraries/ms-common/inc/soft_timer.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
```

Comment:
This defines an API for "one-off" timers, right? For our use cases, do you think it'd be worth it (and beneficial) to implement timers to handle recurring events (do ``x`` every ``y`` sort of thing)? Or did you plan on doing all that work in the callback function, and re-register the timer once it's complete?